### PR TITLE
Product details page

### PR DESCRIPTION
**Pull request by bhtibrewal**
- [ ] filter functionality
- [ ] conditional routing to private routes
- [ ] fetch wishlist , cart when user logs in

### CODE REVIEW

File: src/components/sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            }}
+          />{" "}
+          <span className="body-l">Include Out Of Stock</span>
+        </label>
+        <label>
+          <input
+            type="checkbox"
+            checked={showFastDelivery}
+            onChange={() => {
+              filterStateDispatch({ type: "FAST_DELIVERY" });
```
Comment (pankaj-prog) [inline src/components/sidebar/Sidebar.jsx:35]: you can create constants for actions types.

File: src/components/sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            }}
+          />
+          <span className="body-l">Fast Delivery</span>
+        </label>
+      </div>
+      <h3>Category</h3>
+      <div className="flex-col">
+        {categoriesList?.map((category) => {
+          return (
+            <label key={category.categoryName}>
```
Comment (pankaj-prog) [inline src/components/sidebar/Sidebar.jsx]: you can destructure the category name, as you are using it at multiple places.

File: src/context/ProductContext.jsx [hunk 3]
```diff
@@ -1,41 +1,11 @@
...
-            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI4MmIxNjYwYi0yMTg2LTRiMmEtOTRlOC1mYzNmODUyYjcxNTQiLCJlbWFpbCI6ImpvaG5kb2VAZ21haWwuY29tIiwiaWF0IjoxNjQ3MzQ3OTY1fQ.CsaRbb8b9jWPRPUMY8OIt-TWE1bcRbFiUDz22YbNAoY",
-        },
-      });
-      return res.data.cart;
-    } catch (error) {
-      console.log(`couldn't get cart`, { error });
-    }
-  };
-
   const initialProductState = {
```
Comment (pankaj-prog) [inline src/context/ProductContext.jsx:9]: naming convention should be consistent

File: src/context/ProductContext.jsx [hunk 4]
```diff
@@ -61,7 +36,38 @@ const ProductContextProvider = ({ children }) => {
     product_reducer_fn,
     initialProductState
   );
-  console.log("this is productstate", productState);
+  const { loginState } = useUserContext();
+  const fetchedProductList = useAxios("api/products", "GET", "products");
+
+  const fetchWishlist = useAxios(
+    "/api/user/wishlist",
```
Comment (pankaj-prog) [inline src/context/ProductContext.jsx:43]: this is good that you are using a custom hook for your API calls.

**Review by pankaj-prog — APPROVED** [review]
I learned how to create a custom hook for all the API calls. 
You did awesome work.

**Review by sri-rishi — APPROVED** [review]
Nice work, learnt some new approaches from your code , other than changes mention above couldn't find any changes , feel free to merge

File: src/components/card/BasicCard.jsx [hunk 5]
```diff
@@ -1,25 +1,22 @@
...
       <button className="icon favourite-icon">
         <i className="fa-regular fa-heart fa-2x"></i>
       </button>
       <div className="content">
-        <img
-          className="card-img"
-          src="https://render.fineartamerica.com/images/images-new-artwork/images-medium-5/la-porta-rossa-sulla-salita-guido-borelli.jpg?v=2"
-          alt=""
-        />
+        <img className="card-img" src={img_src} alt="" />
```
Comment (Priya-Patel31) [inline src/components/card/BasicCard.jsx:19]: write something in the alt tag to show alternate text .

File: src/components/card/TextOverMediaCard.jsx [hunk 6]
```diff
@@ -1,25 +1,44 @@
-export const TextOverMediaCard = ({ item }) => {
+import { useProductContext } from "../../context";
+import { RatingPalleteIcon } from "../index";
+export function inWhisList(wishList, art) {
```
Comment (Priya-Patel31) [inline src/components/card/TextOverMediaCard.jsx:3]: typo mistake  wishlist spelling is wrong .

File: src/components/card/TextOverMediaCard.jsx [hunk 7]
```diff
@@ -1,25 +1,44 @@
...
+
   return (
     <div className="card w-30 text-o-media">
       <div className="content">
         <img className="card-img" src={img_src} alt="" />
         <button className="icon favourite-icon">
-          <i className="fa-regular fa-heart fa-2x"></i>
+          <i
+            className={`${
+              // () =>inWhisList(wishList, art) ? "fas" :
```
Comment (Priya-Patel31) [inline src/components/card/TextOverMediaCard.jsx:33]: remove the commented code

File: src/components/card/TextOverMediaCard.jsx [hunk 8]
```diff
@@ -1,25 +1,44 @@
...
+  const { productState } = useProductContext();
+  const { wishList, cart } = productState;
+
   return (
     <div className="card w-30 text-o-media">
       <div className="content">
         <img className="card-img" src={img_src} alt="" />
         <button className="icon favourite-icon">
-          <i className="fa-regular fa-heart fa-2x"></i>
+          <i
```
Comment (Priya-Patel31) [inline src/components/card/TextOverMediaCard.jsx:31]: Suggestion : You can use React Icons.

File: src/components/navbar/components/LoggedInUser.jsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
+
+export const LoggedInUser = () => {
+  return (
+    <div className="user">
+    <div className="avatar-text avatar-s">BT</div>
+    <div className="user-dropdown flex-col">
+      <a
```
Comment (Priya-Patel31) [inline src/components/navbar/components/LoggedInUser.jsx:7]: Replace this a tag with Link tag.

File: src/components/navbar/components/RightNav.jsx [hunk 10]
```diff
@@ -1,47 +1,40 @@
-import { Link } from "react-router-dom";
-import { useUserContext } from "../../../context";
+import { NavLink, useNavigate } from "react-router-dom";
+import { useProductContext, useUserContext } from "../../../context";
+import { LoggedInUser } from "./LoggedInUser";
+
 export const RightNav = ({ onClick }) => {
```
Comment (Priya-Patel31) [inline src/components/navbar/components/RightNav.jsx:5]: from where this onClick is coming up ??

**Review by Priya-Patel31 — APPROVED** [review]
The code looks clean and well structured . Really liked the good practices which you have used .
Just have few minor changes you can go through it

**Pull request merged**